src/Misc/MiddleWare.cpp

```diff
@@ -271,6 +281,79 @@ void preparePadSynth(string path, PADnoteParameters *p, rtosc::RtData &d)
     }
 }
 
+/**
+ * Class responsible for sending "wavetable-params-changed", remembering
+ * where it has been sent and at what time.
+ *
+ * sending chained requests of "/wavetable-params-changed"
+ * TODO WT1: This could be removed, because changes are detected in a loop in
+ *           ADnoteParameters::requestWaveTables (every 0.2s). But this has no
+ *           latency, so it's faster. So I'm for keeping this class.
+ */
+class WaveTableRequestHandler
+{
+public:
+    using param_change_time_t = int;
+private:
+    struct info_t
+    {
+        //! Do not request ad-pars if we're already waiting for them
+        bool waitForAdPars = false;
+        //! Do not generate waves for out-dated parameters.
+        //! This marks which scale tensors are currently valid.
+        //! First time will be "1".
+        param_change_time_t param_change_time = 0;
+    };
+
+    info_t info[NUM_MIDI_PARTS][NUM_KIT_ITEMS][NUM_VOICES][2];
+
+public:
+    //! If not already done, inform RT that params relevant for wavetable
+    //! calculation failed. If freqs or semantics could have changed, send those
```

Comment:
When would the calculations fail?

Reply:
This should be "changed", not "failed". Just used the wrong word.